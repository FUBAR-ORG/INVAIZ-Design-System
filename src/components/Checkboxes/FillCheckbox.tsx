import type { CheckboxDefaultProps } from '@components/Checkboxes/interfaces/Checkbox.interface';
// types

import HoverCircle from '@components/Checkboxes/HoverCircle';
// components

import {
  StyleFillCheckIcon,
  StyleFillCheckbox,
  StyleFillCheckboxInput,
} from '@components/Checkboxes/styles/FillCheckbox.style';
// styles

/**
 * 다중 선택 체크박스
 */
const FillCheckbox = ({ disabled, onChange, ...props }: CheckboxDefaultProps) => (
  <HoverCircle disabled={disabled}>
    <StyleFillCheckboxInput
      type='checkbox'
      onChange={(e) => onChange?.(e.target.checked)}
      disabled={disabled}
      {...props}
    />
    <StyleFillCheckbox>
      <StyleFillCheckIcon size={20} icon='Check' />
    </StyleFillCheckbox>
  </HoverCircle>
);

export default FillCheckbox;
