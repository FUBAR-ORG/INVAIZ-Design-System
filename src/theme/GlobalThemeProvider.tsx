import { PropsWithChildren } from 'react';
import { ThemeProvider } from '@emotion/react';
import font, { fontSize, fontWeight, lineHeight } from '@theme/font';
import color from '@theme/color';

export default function GlobalThemeProvider({ children }: PropsWithChildren<unknown>) {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
}

const theme = { font, fontSize, lineHeight, fontWeight, color };
