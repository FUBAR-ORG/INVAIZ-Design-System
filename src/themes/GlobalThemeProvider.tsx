import type { PropsWithChildren } from 'react';
import type { EmotionTheme } from '@themes/emotion-theme';
import type { ColorTheme } from '@themes/colors/color';
// types

import { ThemeProvider } from '@emotion/react';
// React modules

import font, { fontSize, fontWeight, lineHeight, textAlign, fontColor } from '@themes/font';
import { colorMap, normal, light } from '@themes/colors/color';
import style from '@themes/style';
import GlobalStyle from '@themes/GlobalStyle';

interface GlobalThemeProviderProps {
  colorTheme?: ColorTheme;
}

export default function GlobalThemeProvider({
  children,
  colorTheme = 'normal',
}: PropsWithChildren<GlobalThemeProviderProps>) {
  const theme: EmotionTheme = {
    normal,
    light,

    color: colorMap[colorTheme],

    font,
    fontSize,
    lineHeight,
    fontWeight,
    textAlign,
    fontColor,

    style,
  };
  return (
    <ThemeProvider theme={theme}>
      {children}
      <GlobalStyle />
    </ThemeProvider>
  );
}
