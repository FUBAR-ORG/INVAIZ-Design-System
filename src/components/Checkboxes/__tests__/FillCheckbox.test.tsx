import { render } from '@tests/test-utils';
import { create } from 'react-test-renderer';
// test utils

import FillCheckbox from '@components/Checkboxes/FillCheckbox';
// components

import GlobalThemeProvider from '@themes/GlobalThemeProvider';
// providers

describe('FillCheckbox', () => {
  it('체크박스의 기본값은 `false`이며, 변경이 가능하다.', () => {
    const { getAllByRole } = render(
      <>
        <FillCheckbox />
        <FillCheckbox checked />
      </>
    );
    const [isNotChanged, isChanged] = getAllByRole('checkbox');

    expect(isNotChanged).not.toBeChecked();
    expect(isChanged).toBeChecked();
  });

  it('체크박스는 클릭할 수 있으며, 클릭 시 `checked` 상태 변경이 가능하다.', () => {
    const onChange = jest.fn();
    const { getByRole } = render(<FillCheckbox onChange={onChange} />);
    const checkbox = getByRole('checkbox');

    expect(onChange).not.toHaveBeenCalled();

    checkbox.click();
    expect(onChange).toHaveBeenCalledWith(true);
    expect(checkbox).toBeChecked();

    checkbox.click();
    expect(onChange).toHaveBeenCalledWith(false);
    expect(checkbox).not.toBeChecked();

    expect(onChange).toHaveBeenCalled();
  });

  it('체크박스는 `disabled`일 때, 상태 변경이 불가능하다.', () => {
    const { getByRole } = render(<FillCheckbox disabled />);
    const checkbox = getByRole('checkbox');

    checkbox.click();
    expect(checkbox).not.toBeChecked();
  });
});

test('FillCheckbox Snapshot', () => {
  const checked = create(
    <GlobalThemeProvider>
      <FillCheckbox checked />
    </GlobalThemeProvider>
  );
  expect(checked).toMatchSnapshot();

  const unChecked = create(
    <GlobalThemeProvider>
      <FillCheckbox />
    </GlobalThemeProvider>
  );
  expect(unChecked).toMatchSnapshot();
});
