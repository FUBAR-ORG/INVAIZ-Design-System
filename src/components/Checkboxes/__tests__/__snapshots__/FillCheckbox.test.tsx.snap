// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FillCheckbox Snapshot 1`] = `
<span
  className="css-srlyvc"
>
  <input
    checked={true}
    className="css-10ql71c"
    onChange={[Function]}
    type="checkbox"
  />
  <div
    className="css-1hzayjd"
  >
    <svg
      className="css-1a3jyl"
      data-file-name="SvgCheck"
      height={20}
      width={20}
    />
  </div>
</span>
`;

exports[`FillCheckbox Snapshot 2`] = `
<span
  className="css-srlyvc"
>
  <input
    className="css-10ql71c"
    onChange={[Function]}
    type="checkbox"
  />
  <div
    className="css-1hzayjd"
  >
    <svg
      className="css-1a3jyl"
      data-file-name="SvgCheck"
      height={20}
      width={20}
    />
  </div>
</span>
`;
